src/server/system_services/bucket_server.js

```diff
@@ -1132,6 +1132,26 @@ function update_bucket_lambda_trigger(req) {
         .return();
 }
 
+
+async function update_all_buckets_default_pool(req) {
+    const pool_name = req.rpc_params.pool_name;
+    const pool = req.system.pools_by_name[pool_name];
+    if (!pool) throw new RpcError('INVALID_POOL_NAME');
+    const internal_pool = pool_server.get_internal_mongo_pool(pool.system);
+    if (String(pool._id) === String(internal_pool._id)) return;
+    const buckets_with_internal_pool = _.filter(req.system.buckets_by_name, bucket =>
+        is_using_internal_storage(bucket, internal_pool));
+
+    return P.map(buckets_with_internal_pool, bucket => server_rpc.client.tier.update_tier({
+        name: bucket.tiering.tiers[0].tier.name,
+        attached_pools: [pool_name],
+        data_placement: 'SPREAD'
+    }, {
+        auth_token: req.auth_token
+    })).return();
```

Comment:
Since you are inside an async function you can make it cleaner with await -
just remove this `.return()` and replace the `return P.map(...)` with `await P.map(...);`

Reply:
changed to await in make_changes